### DISCUSSION TITLE
Set window size according to columns in an Emacs window

### DISCUSSION THREAD

**Discussion by plantarum** [i3/i3#4887] [Q&A]
Hi,

I have been using i3wm for years, and for the most part the default behaviours are all that I need. I use windows full screen, or occasionally two windows side-by-side with equal size.

However, sometimes I am reviewing a webpage or pdf document in one window, with a notes file open in emacs in the adjacent window. In this case, I like to have the Emacs window narrowed such that it just shows 80 columns, which allows me to expand the pdf window so I can zoom in on the document a little more.

I do this by dragging the window borders, which is easy to do. But I wonder, is it possible to script this? That is, to query Emacs to find out how wide it needs to be to display 80 columns, then pass that info to i3wm to have it do the autoresizing based on that value?

I could do this with fixed width values, since I'm usually using the same monitor. But for times when I'm working from a different office, or on my laptop screen, it would be nice to have the values set based on the active screen.

Thanks,

Tyler

i3 4.17.1
Ubuntu 20.04
Emacs 29.0.50

**Comment by Airblader** [comment]
There is no generic way of doing this in X, so if emacs doesn't have some API that allows calculating this, the answer is "no". Whether or not emacs has such an API is something I don't know, and (unless you know already), might be better asked in the emacs community circles.

**Comment by orestisfl** [comment]
It might be worth to check the output of xprop on the emacs window as well
